### PR TITLE
Thematize `CircleProgress` component

### LINKED ISSUE #2342: Thematize `CircleProgress` component
### Description

Yamada UI components are tied to a theme. However, the`CircleProgress` component does not have a theme.

### Problem Statement/Justification

Themes have `defaultProps`, which can be utilized to share `props` throughout the project. However, the `FileButton` component does not have a theme, so it cannot take advantage of them.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by cidkumagai**
Closes #2342

### CODE REVIEW

File: packages/components/progress/src/circle-progress.tsx [hunk 1]
```diff
@@ -78,11 +90,15 @@ export type CircleProgressProps = Omit<HTMLUIProps<"div">, "color"> &
  * @see Docs https://yamada-ui.com/components/feedback/circle-progress
  */
 export const CircleProgress = forwardRef<CircleProgressProps, "div">(
-  (
-    {
+  ({ size: boxSize = "6rem", ...props }, ref) => {
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:93]: Please change `({ size: boxSize = "6rem", ...props }, ref) => {` to `(props, ref) => {`.

File: packages/components/progress/src/circle-progress.tsx [hunk 2]
```diff
@@ -78,11 +90,15 @@ export type CircleProgressProps = Omit<HTMLUIProps<"div">, "color"> &
...
+    const [styles, mergedProps] = useComponentStyle("CircleProgress", {
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:94]: Please change

```
    const [styles, mergedProps] = useComponentStyle("CircleProgress", {
      boxSize,
      ...props,
    })
```

to

```
    const [styles, { size = "6em", ...mergedProps }] = useComponentStyle(
      "CircleProgress",
      props,
    )
```

File: packages/components/progress/src/circle-progress.tsx [hunk 3]
```diff
@@ -78,11 +90,15 @@ export type CircleProgressProps = Omit<HTMLUIProps<"div">, "color"> &
...
+      boxSize,
+      ...props,
+    })
+
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:98]: Please remove empty line.

File: packages/components/progress/src/circle-progress.tsx [hunk 4]
```diff
@@ -78,11 +90,15 @@ export type CircleProgressProps = Omit<HTMLUIProps<"div">, "color"> &
...
+    let {
       className,
       children,
-      size = "6rem",
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:85]: Please change `size = "6rem",` to `boxSize,`.

File: packages/components/progress/src/circle-progress.tsx [hunk 5]
```diff
@@ -127,13 +144,11 @@ export const CircleProgress = forwardRef<CircleProgressProps, "div">(
     })
 
     const css: CSSUIObject = {
-      display: "inline-block",
-      position: "relative",
-      verticalAlign: "middle",
-      fontSize: size,
+      ...styles,
+      boxSize,
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:148]: Please `boxSize,` to `fontSize: boxSize,`.

File: packages/components/progress/src/circle-progress.tsx [hunk 6]
```diff
@@ -93,10 +109,11 @@ export const CircleProgress = forwardRef<CircleProgressProps, "div">(
       isRounded,
       speed = ["1.4s", "2s"],
       ...rest
-    },
-    ref,
-  ) => {
-    size = (useToken("sizes", useValue(size)) ?? size) as string | number
+    } = omitThemeProps(mergedProps)
+
+    boxSize = (useToken("sizes", useValue(boxSize)) ?? boxSize) as
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:114]: Please change

```
    boxSize = (useToken("sizes", useValue(boxSize)) ?? boxSize) as
      | string
      | number
```

to

```
    boxSize ??= replaceObject(size, (value) =>
      !isUnit(value) ? useToken("sizes", value) : value,
    )
```

File: packages/components/progress/src/circle-progress.tsx [hunk 7]
```diff
@@ -93,10 +109,11 @@ export const CircleProgress = forwardRef<CircleProgressProps, "div">(
...
+      | string
+      | number
     thickness = (useToken("sizes", useValue(thickness)) ?? thickness) as
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:117]: Please change

```
    thickness = (useToken("sizes", useValue(thickness)) ?? thickness) as
      | string
      | number
```

to

```
    thickness = replaceObject(thickness, (value) =>
      !isUnit(value) ? useToken("sizes", value) : value,
    )
```

File: packages/components/progress/src/circle-progress.tsx [hunk 8]
```diff
@@ -127,13 +144,11 @@ export const CircleProgress = forwardRef<CircleProgressProps, "div">(
...
-      display: "inline-block",
-      position: "relative",
-      verticalAlign: "middle",
-      fontSize: size,
+      ...styles,
+      boxSize,
     }
 
-    const props: CircleProps = isAnimation
+    const restProps: CircleProps = isAnimation
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:151]: Please change `restProps` to `circleProps`.

File: packages/components/progress/src/circle-progress.tsx [hunk 9]
```diff
@@ -153,14 +168,14 @@ export const CircleProgress = forwardRef<CircleProgressProps, "div">(
         __css={css}
         {...rest}
       >
-        <Shape size={size} isAnimation={isAnimation} speed={speed}>
+        <Shape size={boxSize} isAnimation={isAnimation} speed={speed}>
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:171]: Please change `size={boxSize}` to `boxSize={boxSize}`.

File: packages/components/progress/src/circle-progress.tsx [hunk 10]
```diff
@@ -153,14 +168,14 @@ export const CircleProgress = forwardRef<CircleProgressProps, "div">(
...
           <Circle stroke={trackColor} strokeWidth={thickness} />
           <Circle
             stroke={color}
             strokeWidth={thickness}
             strokeLinecap={isRounded ? "round" : undefined}
             opacity={isTransparent ? 0 : undefined}
-            {...props}
+            {...restProps}
```
Comment (hirotomoyamada) [inline packages/components/progress/src/circle-progress.tsx:178]: Please change `{...restProps}` to `{...circleProps}`.

**Review by hirotomoyamada — COMMENTED** [review]
Please change

```diff
type ShapeProps = Omit<HTMLUIProps<"svg">, "speed"> &
  Pick<
    Required<CircleProgressProps>,
-    "children" | "size" | "isAnimation" | "speed"
+    "children"  | "isAnimation" | "speed"
  >
- const Shape: FC<ShapeProps> = ({ size, isAnimation, speed, ...rest }) => {
+ const Shape: FC<ShapeProps> = ({ boxSize, isAnimation, speed, ...rest }) => {
  const animation = useAnimation({
    keyframes: {
      "0%": {
        transform: "rotate(0deg)",
      },
      "100%": {
        transform: "rotate(360deg)",
      },
    },
    duration: typeof speed[1] === "string" ? speed[1] : `${speed[1]}s`,
    iterationCount: "infinite",
    timingFunction: "linear",
  })
  const css: CSSUIObject = {
    display: "block",
-    boxSize: size,
+    boxSize,
    ...(isAnimation ? { animation } : {}),
  }
  return <ui.svg viewBox="0 0 100 100" __css={css} {...rest} />
}
```

**Review by hirotomoyamada — COMMENTED** [review]
Change the Storybook as well.

**Comment by cidkumagai** [comment]
@hirotomoyamada 

Thank you for review.
I’ve made the revisions, so please review them.

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**